app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/nutrition/NutritionProductFragment.kt

```diff
@@ -324,7 +324,13 @@ class NutritionProductFragment : BaseFragment(), CustomTabActivityHelper.Connect
             nutrimentListItems.add(BoldNutrimentListItem(getString(R.string.nutrition_minerals)))
             nutrimentListItems.addAll(getNutrimentItems(nutriments, Nutriments.MINERALS_MAP))
         }
-        binding.nutrimentsRecyclerView.adapter = NutrimentsGridAdapter(nutrimentListItems)
+
+        // Show nutrition table and nutrition per portion button if nutritional values are available
+        if (nutrimentListItems.size > 1) {
```

Comment:
```suggestion
        if (nutrimentListItems.isNotEmpty()) {
```

Reply:
@VaiTon Here, the size of `nutrimentListItems` is 1 even if no values are present. So, it will never be empty. Therefore, the changes you are asking are not valid. There must be at least 2 elements in the arrayList, then only we must set adapter to recyclerView.